### PR TITLE
Some improvement on batched series observation call; Support fetch USA cities

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

**Review by pradh — APPROVED** [review]
Nice!

File: server/routes/shared_api/observation/series.py [hunk 1]
```diff
@@ -68,34 +73,44 @@ def series_all():
...
+
   variables = list(filter(lambda x: x != "", request.args.getlist('variables')))
   if not variables:
     return 'error: must provide a `variables` field', 400
+
   facet_ids = list(filter(lambda x: x != "", request.args.getlist('facetIds')))
-  batch_size = request.args.get('batchSize') or _MAX_BATCH_SIZE
+
   # Make batched calls there are too many child places for server to handle
+  batch_size = _MAX_BATCH_SIZE // len(variables)
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/shared_api/observation/series.py:98]: Done

File: server/routes/shared_api/observation/series.py [hunk 2]
```diff
@@ -68,34 +73,44 @@ def series_all():
...
-      child_place_batches = list(
-          shared.divide_into_batches(child_places, batch_size))
       merged_response = {}
-      for batch in child_place_batches:
+      batch_count = 1
+      for batch in shared.divide_into_batches(child_places, batch_size):
+        logging.info(
+            f"Batch {batch_count} of {math.ceil(len(child_places) / batch_size)}"
+        )
         new_response = fetch.series_core(batch, variables, False, facet_ids)
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/shared_api/observation/series.py:106]: Only keeping the initial logging now.

**Review by shifucun — COMMENTED** [review]
thanks for review!

**Pull request merged**